agora/footfall_ai_api/src/video_processor.py

```diff
@@ -73,28 +77,38 @@ def start(self):
         if not self.running:
             self.running = True
             self.vs = VideoCapture(self.url, self.skip_fps, queue_size=150)
-            self.process_thread = threading.Thread(target=self.process_frames)
-            self.process_thread.start()
+            self.process_thread = thread_pool.submit(self.process_frames)
             print(f"Started processing thread for video {self.index}")
 
     def stop(self):
         self.running = False
         if self.process_thread:
-            self.process_thread.join()
+            self.process_thread.result()  # Wait for the thread to finish
         if self.vs:
             self.vs.stop()
+        # Clear the queue to free up memory
+        with self.processed_frame_queue.mutex:
+            self.processed_frame_queue.queue.clear()
         print(f"Stopped processing thread for video {self.index}")
 
     def process_frames(self):
         processing_times = collections.deque(maxlen=200)
         frame_count = 0
         last_time = time.time()
+        original_width, original_height = 640, 360  # Original resolution
+        new_width, new_height = 320, 180  # New resolution
+        scaling_factor_x = new_width / original_width
+        scaling_factor_y = new_height / original_height
+
         while self.running:
             frame, success = self.vs.read()
-            original_frame = frame.copy()
             if not success:
                 continue
 
+            # Reduce input resolution
+            frame = cv2.resize(frame, (new_width, new_height))
```

Comment:
What about the new frame size? Does the model detects people with such a small size?

Reply:
agree